Userland/Libraries/LibWeb/Animations/KeyframeEffect.cpp

```diff
@@ -730,29 +730,48 @@ void KeyframeEffect::set_target(DOM::Element* target)
         m_target_element->associate_with_effect(*this);
 }
 
-void KeyframeEffect::set_pseudo_element(Optional<String> pseudo_element)
+WebIDL::ExceptionOr<void> KeyframeEffect::set_pseudo_element(Optional<String> pseudo_element)
 {
+    auto& realm = this->realm();
+
     // On setting, sets the target pseudo-selector of the animation effect to the provided value after applying the
     // following exceptions:
 
     // FIXME:
-    // - If the provided value is not null and is an invalid <pseudo-element-selector>, the user agent must throw a
-    //   DOMException with error name SyntaxError and leave the target pseudo-selector of this animation effect
-    //   unchanged.
-
     // - If one of the legacy Selectors Level 2 single-colon selectors (':before', ':after', ':first-letter', or
     //   ':first-line') is specified, the target pseudo-selector must be set to the equivalent two-colon selector
     //   (e.g. '::before').
     if (pseudo_element.has_value()) {
         auto value = pseudo_element.value();
 
         if (value == ":before" || value == ":after" || value == ":first-letter" || value == ":first-line") {
-            m_target_pseudo_selector = MUST(String::formatted(":{}", value));
-            return;
+            m_target_pseudo_selector = CSS::Selector::PseudoElement::from_string(MUST(value.substring_from_byte_offset(1)));
+            return {};
         }
     }
 
-    m_target_pseudo_selector = pseudo_element;
+    // - If the provided value is not null and is an invalid <pseudo-element-selector>, the user agent must throw a
+    //   DOMException with error name SyntaxError and leave the target pseudo-selector of this animation effect
+    //   unchanged.
+    if (pseudo_element.has_value()) {
+        auto pseudo_element_without_colons = MUST(pseudo_element->replace("::"sv, ""sv, ReplaceMode::FirstOnly));
```

Comment:
Yeah I was looking for a `substring_after` sort of function but didn't find one, adding something like that might be nice. I think I'll leave it like this for now, since this should be a rare code path anyways.